internal/service/datasync/location_efs.go

```diff
@@ -42,13 +48,16 @@ func ResourceLocationEFS() *schema.Resource {
 							Type:     schema.TypeSet,
 							Required: true,
 							ForceNew: true,
-							Elem:     &schema.Schema{Type: schema.TypeString},
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: verify.ValidARN,
+							},
 						},
 						"subnet_arn": {
 							Type:         schema.TypeString,
 							Required:     true,
 							ForceNew:     true,
-							ValidateFunc: validation.NoZeroValues,
+							ValidateFunc: verify.ValidARN,
```

Comment:
these are changes to code logic not mentioned in the pr (as well as e.g. the %s/%w replacements) - I am not quite sure if they should be included within this pr